### PR TITLE
Features Endpoint: Permissions to view stats, map items, see versions

### LINKED ISSUE #2969: Features endpoint - Permissions to view stats, map items, see versions
**Is your feature request related to a problem? Please describe.**
DSpace 6 statistics links are only displayed depending permissions configured in https://github.com/DSpace/DSpace/blob/main/dspace/config/modules/usage-statistics.cfg#L26-L31, and are not always accessible to all
DSpace 6 item mapping features is only available if the user has ADD and WRITE permissions to the current collection
DSpace 6 versioning details are only displayed if configured in https://github.com/DSpace/DSpace/blob/main/dspace/config/modules/versioning.cfg#L11

**Describe the solution you'd like**
Based on https://docs.google.com/spreadsheets/d/1182LcD_WqIZRbUGWpLtBw0aOMR9jhbOVB7GZqtTpR9A/edit#gid=0, implement the features:
* canViewUsageStatistics
* canViewWorkflowStatistics
* canViewSearchStatistics
* canManageMappedItems
* canViewVersions

### PR DESCRIPTION
**Pull request by Raf-atmire**
## References
* Fixes https://github.com/DSpace/DSpace/issues/2969

## Description
This PR introduces new REST Authorization features
- canManageMappedItems
- canViewSearchStatistics
- canViewUsageStatistics
- canViewVersions
- canViewWorkflowStatistics

## Instructions for Reviewers

List of changes in this PR:
* Added the five above mentioned features for various DSpaceObjects
* Provided tests for these new features

How to test this PR:
- Retrieve the features for the Item, Collection, Community and Site objects as an admin and verify that their respective new features are present. The tests will deal with all cases that can happen for each feature individually.

## Checklist

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

File: dspace-api/src/test/data/dspaceFolder/config/local.cfg [hunk 1]
```diff
@@ -129,3 +129,11 @@ configuration.exposed.array.value = public_value_1, public_value_2
...
 authentication-ip.Staff = 5.5.5.5
 authentication-ip.Student = 6.6.6.6
+
+# Test config for the usage statistics authorization
+usage-statistics.authorization.admin.usage = true
+usage-statistics.authorization.admin.workflow = true
+usage-statistics.authorization.admin.search = true
+
+# Test config for the versioning authorization
+versioning.item.history.view.admin = false
```
Comment (tdonohue) [inline dspace-api/src/test/data/dspaceFolder/config/local.cfg:139]: These settings are all unnecessary, as they are the default values already in `versioning.cfg` and `usage-statistics.cfg`.

In any case, you also do NOT need to add most settings to this `local.cfg`.  Settings for Unit/Integration tests can be tweaked dynamically in the tests themselves by calling `configurationService.setProperty()` to change the value to whatever you want it to be.  I see you've done this already in some of your ITs, which is correct.

Configurations ONLY need to be added to this `local.cfg` if they CANNOT be reloaded/set dynamically.  There are very few configurations that fall into this category...mostly those that initialize our database/webapp.

In any case, all the above is just an FYI. These changes can be removed.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@Raf-atmire : Overall, this looks correct.  Just a minor request below to remove unnecessary configurations from your test environment.  Beyond that, I'm +1 this PR.

Again, the new code here is very small ~350 lines of new code, much of it comments.  Most of the code here is new ITs, which I've code reviewed.  So, flagging this as `1 Approval`.

File: dspace-api/src/test/data/dspaceFolder/config/local.cfg (hunk 1 above)
Comment (Raf-atmire) [reply dspace-api/src/test/data/dspaceFolder/config/local.cfg]: @tdonohue Thank you for the feedback, I've removed the unnecessary config parameters in the test local.cfg file

**Review by tdonohue — APPROVED** [review]
👍 Looks good to me now. Thanks @Raf-atmire !

**Comment by tdonohue** [comment]
Merging as we agreed to `1 approval` in today's meeting.

**Pull request merged**